charts/karpenter/templates/deployment.yaml

```
@@ -89,6 +89,10 @@ spec:
           {{- with .Values.controller.env }}
             {{- toYaml . | nindent 12 }}
           {{- end }}
+          envFrom:
+          {{- with .Values.controller.envFrom }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
```

I think this should be like below so that it works with the default empty list:

```suggestion
          {{- with .Values.controller.envFrom }}
          envFrom:
            {{- toYaml . | nindent 12 }}
          {{- end }}
```